pkg/repository/prisma/workflow_run.go

```diff
@@ -725,354 +753,432 @@ func workflowRunMetricsCount(ctx context.Context, pool *pgxpool.Pool, queries *d
 	return workflowRunsCount, nil
 }
 
-func createNewWorkflowRun(ctx context.Context, pool *pgxpool.Pool, queries *dbsqlc.Queries, l *zerolog.Logger, tenantId string, opts *repository.CreateWorkflowRunOpts) (*dbsqlc.WorkflowRun, error) {
-	ctx, span := telemetry.NewSpan(ctx, "db-create-new-workflow-run")
+func createNewWorkflowRuns(ctx context.Context, pool *pgxpool.Pool, queries *dbsqlc.Queries, l *zerolog.Logger, tenantId string, inputOpts []*repository.CreateWorkflowRunOpts) ([]*dbsqlc.WorkflowRun, error) {
```

Comment:
@grutt @abelanger5 

I'd appreciate some eyes on this. I'm still unsure about a couple of parts of it. The ParentId stuff I'm still confused by so.
I've removed the dedupe stuff.

I can run it locally and I can create workflows and whatnot. I'd appreciate help creating test cases to verify things are working well.
I can get a bunch of deadlocks when I try to increase past 200 workflow runs but I'm guessing it could just be contention in my db because smaller sizes seem to work fine.

Reply:
i want to reread one more time, but this looks reasonable